src/com/ms/silverking/io/StreamUtil.java

```diff
@@ -15,265 +13,264 @@
+    StringBuilder sb;
+    int c;
+
+    sb = new StringBuilder();
+    do {
+      c = in.read();
+      if (c != NEWLINE && c != CR && c >= 0) {
+        sb.append((char) c);
+      }
+    } while (c != NEWLINE && c != CR && c >= 0);
+    return sb.toString();
+  }
+
+  public static byte[] readToBytes(InputStream in, boolean close) throws IOException {
+    ByteArrayOutputStream tmp;
+
+    tmp = new ByteArrayOutputStream();
+    stream(in, tmp, close, new byte[DEF_BUF_SIZE]);
+    return tmp.toByteArray();
+  }
+
+  ////////////////////////////////////////////////////////////
+
+  public static long limitedStream(InputStream in, OutputStream out, boolean close, byte[] buf, long limit)
+      throws IOException {
+    long numWritten;
+
+    numWritten = 0;
+    try {
+      int numRead;
+
+      numRead = 0;
+      do {
+        int maxToRead;
+        long remaining;
+
+        remaining = limit - numWritten;
+        if (remaining >= buf.length) {
+          maxToRead = buf.length;
+        } else {
+          maxToRead = (int) remaining;
+        }
+        numRead = in.read(buf, 0, maxToRead);
+        if (numRead > 0) {
+          out.write(buf, 0, numRead);
+          out.flush();
+          numWritten += numRead;
+        }
+      } while (numRead >= 0 && numWritten < limit);
+    } finally {
+      out.flush();
+      if (close && in != null) {
         try {
-            int    numRead;
-            
-            numRead = 0;
-            do {
-                int        maxToRead;
-                long    remaining;
-                
-                remaining = limit - numWritten;
-                if (remaining >= buf.length) {
-                    maxToRead = buf.length;
-                } else {
-                    maxToRead = (int)remaining;
-                }
-                numRead = in.read(buf, 0, maxToRead);
-                if (numRead > 0) {
-                    out.write(buf, 0, numRead);
-                    out.flush();
-                    numWritten += numRead;
-                }
-            } while (numRead >= 0 && numWritten < limit);
-        } finally {
-            out.flush();
-            if (close && in != null) {
-                try {
-                    in.close();
-                } catch (IOException ioe_close) {
-                    ioe_close.printStackTrace();
-                }
-            }
-            if (close && out != null) {
-                try {
-                    out.close();
-                } catch (IOException ioe_close) {
-                    ioe_close.printStackTrace();
-                }
-            }
+          in.close();
+        } catch (IOException ioe_close) {
+          ioe_close.printStackTrace();
         }
-        return numWritten;
-    }        
-    
-    
-    public static long stream(InputStream in, OutputStream out, boolean close, byte[] buf) throws IOException {
-        long    numWritten;
-        
-        numWritten = 0;
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 191:197, 198:204, 227:233, 234:240.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.